src/types/NodeType.ts

```diff
@@ -1,80 +1,110 @@
-/* @flow */
 /* eslint-disable no-underscore-dangle */
 
-import { GraphQLObjectType, GraphQLString } from 'graphql';
-import type {
+import {
   GraphQLFieldConfig,
   GraphQLFieldConfigMap,
   GraphQLFieldResolver,
+  GraphQLObjectType,
   GraphQLObjectTypeConfig,
   GraphQLResolveInfo,
+  GraphQLString,
+  Thunk,
 } from 'graphql';
 import {
   connectionDefinitions,
   fromGlobalId,
   globalIdField,
   nodeDefinitions,
 } from 'graphql-relay';
+import invariant from 'invariant';
 import camelCase from 'lodash/camelCase';
 import snakeCase from 'lodash/snakeCase';
 import pluralize from 'pluralize';
-import invariant from 'invariant';
+import { Class } from 'utility-types';
 
+import { Endpoint } from '../resources/HttpResource';
+import Resource from '../resources/Resource';
 import asType from '../utils/asType';
 import resolveThunk from '../utils/resolveThunk';
-import Resource from '../resources/Resource';
 
 type Context = any;
 
+type Args = Record<string, unknown>;
+
 // graphql overrides
 
-type ObjStub = {
-  id: string,
-};
+interface ObjStub {
+  id: string;
+}
 
-type NodeTypeConfig<R: Resource<*>> = {|
-  ...GraphQLObjectTypeConfig<ObjStub, Context>,
-  localIdFieldName?: ?string,
-  Resource: Class<R>,
-  resourceConfig?: mixed,
-|};
+interface ResourceConfig {
+  [key: string]: unknown;
+  endpoint?: Endpoint;
+}
 
-export type NodeFieldResolver<TContext> = GraphQLFieldResolver<
-  ObjStub,
+type NodeTypeConfig<
+  TObject extends ObjStub,
   TContext,
->;
-export type NodeFieldConfig<TContext> = GraphQLFieldConfig<ObjStub, TContext>;
-export type NodeFieldConfigMap<TContext> = GraphQLFieldConfigMap<
-  ObjStub,
-  TContext,
->;
+  R extends Resource<TContext>
+> = GraphQLObjectTypeConfig<TObject, TContext> & {
+  localIdFieldName?: string | null | undefined;
+  Resource: Class<R>;
+  resourceConfig?: ResourceConfig;
+};
+
+export type NodeFieldResolver<
+  TObject extends ObjStub,
+  TContext
+> = GraphQLFieldResolver<TObject, TContext>;
+
+export type NodeFieldConfig<
+  TObject extends ObjStub,
+  TContext
+> = GraphQLFieldConfig<TObject, TContext>;
 
-export type FieldNameResolver<TContext> = (
-  fieldName: string,
-  obj: ObjStub,
-  args: mixed,
+export type NodeFieldConfigMap<
+  TObject extends ObjStub,
+  TContext
+> = GraphQLFieldConfigMap<TObject, TContext>;
+
+export type FieldNameResolver<
+  TObject extends ObjStub,
+  TContext,
+  TArgs = Args
+> = (
+  fieldName: any,
+  obj: TObject,
+  args: TArgs,
   context: TContext,
   info: GraphQLResolveInfo,
-) => string;
+) => any;
```

Comment:
@jquense got most of the types to work correctly save for this. I  had to set `fileName` and the return to `any` due to 

```
src/types/NodeType.ts:230:7 - error TS2345: Argument of type 'string' is not assignable to parameter of type 'keyof TObject'.
  Type 'string' is not assignable to type '"id"'.

230       fieldNameResolver(info.fieldName as any, obj as any, args as any, context as any, info as any),
          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


Found 1 error.
```

Reply:
Why do we even need this `fieldNameResolver`? @taion

Reply:
i think we wanted to make the camel-case logic customizable... but then later on it didn't work anyway

this is vestigial and a quick github search suggests neither bfly nor q-si use this, so we're safe to get rid of `fieldNameResolver`